src/share.ts

```diff
@@ -403,17 +434,32 @@ export async function createShareableUrl(
   }
   logger.debug(`New eval ID on remote instance: ${evalId}`);
 
-  const { domain } = determineShareDomain(evalRecord);
+  return getShareableUrl(evalRecord, showAuth);
+}
 
-  // For custom self-hosted setups, ensure we're using the same domain as the API
-  const customDomain = getEnvString('PROMPTFOO_REMOTE_APP_BASE_URL');
-  const finalDomain = customDomain || domain;
+/**
+ * Checks whether an eval has been shared.
+ * @param eval_ The eval to check.
+ * @returns True if the eval has been shared, false otherwise.
+ */
+export async function hasEvalBeenShared(eval_: Eval): Promise<boolean> {
+  try {
+    const { url } = await getApiConfig(eval_);
 
-  const fullUrl = cloudConfig.isEnabled()
-    ? `${finalDomain}/eval/${evalId}`
-    : SHARE_VIEW_BASE_URL === DEFAULT_SHARE_VIEW_BASE_URL && !customDomain
-      ? `${finalDomain}/eval/${evalId}`
-      : `${finalDomain}/eval/?evalId=${evalId}`;
+    const headers: Record<string, string> = { 'Content-Type': 'application/json' };
 
-  return showAuth ? fullUrl : stripAuthFromUrl(fullUrl);
+    if (cloudConfig.isEnabled()) {
+      headers['Authorization'] = `Bearer ${cloudConfig.getApiKey()}`;
+    }
+
+    const response = await fetchWithProxy(`${url}/${eval_.id}`, {
+      method: 'GET',
+      headers,
+    });
+
+    return response.status !== 404;
```

Comment:
Would like Chief @sklein12 to take a look at this. 

I think I would prefer a new endpoint over checking the status. 

Sharing should probably become an update operation if it's already been shared instead of blocking it.

Reply:
What update is necessary?

Reply:
For when people grade evals and re-share which is a not uncommon use case